cdap-app-templates/cdap-data-quality/src/test/java/co/cask/cdap/dq/testclasses/StreamBatchSource.java

```diff
@@ -115,6 +120,37 @@ public void prepareRun(BatchSourceContext context) {
     context.setInput(stream);
   }
 
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    Arguments arguments = context.getArguments();
+    String sourceDsName = arguments.get("source");
+    String sinkDsName = arguments.get("sink");
+    MetadataEntity sourceEntity = MetadataEntity.ofDataset(sourceDsName);
+    MetadataEntity sinkEntity = MetadataEntity.ofDataset(sinkDsName);
+
+    // propagate all source metadata to sink
+    propogateMetadata(context, sourceEntity, sinkEntity);
+
+    // Propagate metadata of some fields
+    MetadataEntity sourceFieldEntity = sourceEntity.append("field", "empName");
+    MetadataEntity sinkFieldEntity = sinkEntity.append("field", "empName");
+    propogateMetadata(context, sourceFieldEntity, sinkFieldEntity);
+
+    for (Schema.Field field : context.getInputSchema().getFields()) {
+      sourceFieldEntity = sourceEntity.append("field", field.getName());
+      Metadata metadata = context.getMetadata(MetadataScope.USER, sourceFieldEntity);
```

Comment:
Should it be necessary to specify Scope.USER? Or isn't it true that the user can only manipulate USER metadata? What is an example of Scope.SYSTEM metadata?

Reply:
User's can get both system and/or user metadata although they can only add/manipulate metadata in User scope. There is another api which does not take a scope and return metadata for both the scopes.

Reply:
I have moved them to StageContext as needed later on during the implementation we can add implementations/overrides wherever needed.